sources/httpUtils.ts

```diff
@@ -23,6 +25,19 @@ export async function fetchUrlStream(url: string, options: RequestOptions = {})
   });
 }
 
+let globalAgentBootstrapped = false;
+
+async function setupProxy() {
+  if (globalAgentBootstrapped) {
```

Comment:
Maybe this is not required if we know for sure that corepack will execute only one request in process lifetime.